### ISSUE TITLE
Error after FBSDK upgrade to 5.0.0 in xcode 10.1

### ISSUE THREAD

**Issue by luminkhant** [facebook/facebook-swift-sdk#464]
After I had upgrade my FBSDK to version 5.0.0 in pod file as below.
`pod 'FacebookSDK', '~> 5.0.0'
  pod 'FacebookSDK/LoginKit', '~> 5.0.0'
  pod 'FacebookSDK/ShareKit', '~> 5.0.0'
  pod 'FacebookSDK/PlacesKit', '~> 5.0.0'`

I am having this error after pod update. 
`Bolt.h file not found `.
Thus I removed that imported Bolt file and I am having this new error in FBSDK Login Kit.

[https://stackoverflow.com/questions/56421155/getting-error-after-pod-update-to-fbsdk-5-0-0-in-x-code-10-1/56421237#56421237](url)

- Xcode Version: `10.1.0`
- Swift Version: `4`

**Comment by Dfmarulanda** [comment]
Downgrade. They have a lot of problems with the latest release, but not solution in 1 .5 month.

**Comment by luminkhant** [comment]
@Dfmarulanda  Which version would be most stable?

**Comment by fermoya** [comment]
Hi @luminkhant ,

I updated my pods and had the same issue when facebook frameworks updated to:

```
Installing FBSDKCoreKit (5.0.0)
Installing FBSDKLoginKit (5.0.2)
Installing FBSDKMarketingKit (5.0.0)
Installing FBSDKMessengerShareKit (1.3.2)
Installing FBSDKPlacesKit (5.0.2)
Installing FBSDKShareKit (5.0.2)
Installing FacebookSDK (5.0.2)
```

What I did is I unlocked the affected files in my Pods. In those files, I added at the very top, where the imports are, the following line:

`#import "FBSDKInternalUtility.h"`

and substituted:

`[FBSDKBasicUtility objectForJSONString ...]`

for 

`[FBSDKInternalUtility objectForJSONString`

It seems to work and I could log in using Facebook. I hope they fix it soon though. Notice that with the new versions, all

`import FacebookCore`

become

`import FBSDKCoreKit`

same for LoginKit, ShareKit, ...

`SDKApplicationDelegate` renames to `ApplicationDelegate`

Basically, they're asking you to change from their [Swift compiled Pod](https://github.com/facebook/facebook-swift-sdk) to their [Objective-C](https://github.com/facebook/facebook-objc-sdk) one, I don't understand why. There might be some other changes depending on your implementation. 

I hope it works for you too.

**Comment by tayseer5** [comment]
I faced the same issue but it fixed when I 

import FBSDKLoginKit

 
now it work with me

**Comment by rizwan95** [comment]
Somebody fix this bug. It is breaking most of the projects.

**Comment by luminkhant** [comment]
> I faced the same issue but it fixed when I
> 
> import FBSDKLoginKit
> 
> now it work with me

Where did you import that file?

**Comment by luminkhant** [comment]
Here I got the solution. 
https://stackoverflow.com/a/56629791/5149194

**Comment by Shavneet** [comment]
Yes we should try to fix this bug asap  ,  It is blocking  a lot of projects .I think we try to discuss this with each other and try to resolve this our self . If anyone liked the suggestion and ready to do something about this please try to create a medium for live discussion .
Right now i cant think about anything else .

**Comment by rizwan95** [comment]
They should have updated the documentation properly.

**Comment by voltairebiton** [comment]
for those who are using capacitor , the answer from this [link](https://github.com/jeduan/cordova-plugin-facebook4/issues/776#issuecomment-514267272) fixed the issue

**Comment by joesus** [comment]
Please see #433 for reasons on why this happened and how to fix it? Going forward we are committed to preventing breaking changes without proper semantic versioning.
